### PR TITLE
Make hash function on bytes hash the raw bytes

### PR DESCRIPTION
**Pull request by UlfNorell**
For AEVM we do this in the compiler (we don't have the size of the
byte array at runtime). PR: https://github.com/aeternity/aesophia/pull/146

### CODE REVIEW

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 1]
```diff
@@ -4579,6 +4582,52 @@ sophia_crypto(_Cfg) ->
...
+
+            ResSha3_B1    = ?call(call_contract, Acc, IdC, sha3_b17,    word, {Bytes1}),
+            ResSha256_B1  = ?call(call_contract, Acc, IdC, sha256_b17,  word, {Bytes1}),
+            ResBlake2b_B1 = ?call(call_contract, Acc, IdC, blake2b_b17, word, {Bytes1}),
+
+            ResSha3_B2    = ?call(call_contract, Acc, IdC, sha3_b52,    word, {Bytes2}),
+            ResSha256_B2  = ?call(call_contract, Acc, IdC, sha256_b52,  word, {Bytes2}),
+            ResBlake2b_B2 = ?call(call_contract, Acc, IdC, blake2b_b52, word, {Bytes2}),
+
+            ?assertMatchAEVM1OOG(Sha3_B1_N   , ResSha3_B1),
```
Comment (hanssv) [inline apps/aecontract/test/aecontract_SUITE.erl]: could be just one sequence of `?assertMatchVM`?

File: test/contracts/sophia_2/crypto.aes [hunk 2]
```diff
@@ -15,6 +15,14 @@ contract CryptoTest =
   function sha256  (x : complex) : hash = Crypto.sha256(x)
   function blake2b (x : complex) : hash = Crypto.blake2b(x)
 
+  function sha3_b17    (x : bytes(17)) : hash = Crypto.sha3(x)
```
Comment (hanssv) [inline test/contracts/sophia_2/crypto.aes]: Is this one necessary? Didn't the tests exclude minerva?

Comment (UlfNorell) [reply test/contracts/sophia_2/crypto.aes]: Huh, how does it compile on minerva in that case..? 🤔

File: apps/aecontract/test/aecontract_SUITE.erl (hunk 1 above)
Comment (UlfNorell) [reply apps/aecontract/test/aecontract_SUITE.erl]: Yeah I guess. I copied the existing pattern.

File: test/contracts/sophia_2/crypto.aes (hunk 2 above)
Comment (hanssv) [reply test/contracts/sophia_2/crypto.aes]: No, I just meant the bit with the newly added functions?

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 3]
```diff
@@ -4633,6 +4655,52 @@ sophia_crypto(_Cfg) ->
     ?assertMatchFATE({bytes, Sha256},    ResSha256),
     ?assertMatchFATE({bytes, Blake2b},   ResBlake2b),
 
+    %% Tests for bytes (which should hash as raw strings from lima)
+
+    case vm_version() >= ?VM_AEVM_SOPHIA_3 of   %% No bytes in minerva
```
Comment (tolbrino) [inline apps/aecontract/test/aecontract_SUITE.erl]: This looks like it could make use of `?skipRest` to reduce the nesting.

Comment (UlfNorell) [reply apps/aecontract/test/aecontract_SUITE.erl]: There are tests below that shouldn't be skipped...

**Pull request merged**